### PR TITLE
[SOFT-132] Main Event Generator

### PR DESCRIPTION
**Pull request by ArshanKhanifar**
Hello guys, this is the main event generator module, I'll have a call later to explain how it comes into play with the rest of centre console. For now, please take a look and point out any major issues you find with it :) thanks

### CODE REVIEW

File: projects/centre_console/test/test_speed_monitor.c [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  speed = STATIONARY_VELOCITY_THRESHOLD - 20.0;
+  speed_left = (uint16_t)speed, speed_right = (uint16_t)speed;
+  CAN_TRANSMIT_MOTOR_VELOCITY(speed_left, speed_right);
+  MS_TEST_HELPER_CAN_TX_RX(TEST_CAN_EVENT_TX, TEST_CAN_EVENT_RX);
+
+  TEST_ASSERT_EQUAL(*get_global_speed_state(), SPEED_STATE_STATIONARY);
+}
+
+void tset_speed_monitor_has_timeout(void) {
```
Comment (copperium) [inline projects/centre_console/test/test_speed_monitor.c:66]: Typo: `tset`

Comment (ArshanKhanifar) [reply projects/centre_console/test/test_speed_monitor.c:66]: whoa this is BIG, this test didn't even RUN!!!!

File: projects/centre_console/src/main_event_generator.c [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  }
+  if (output_event == NUM_CENTRE_CONSOLE_POWER_EVENTS) {
+    return false;
+  }
+  event_raise(output_event, 0);
+  return true;
+}
+
+#define prv_power_main_or_return(storage)                                         \
+  const PowerState power_state = power_fsm_get_current_state(storage->power_fsm); \
```
[the remark this answers is not in the document]
Comment (ArshanKhanifar) [reply projects/centre_console/src/main_event_generator.c:49]: good point! I didn't care much though cuz it's a private macro only used in this file...

[the remark this answers is not in the document]
Comment (ArshanKhanifar) [reply projects/centre_console/src/main_event_generator.c:49]: changed it anyways.

**Pull request merged**